### PR TITLE
Created an adapter for South Africa

### PR DESCRIPTION
**Pull request by magsyg**
Created an adapter for South Africa, with data from http://saaqis.environment.gov.za/Report/HourlyReports.

This source has PM2.5, PM10, So2, No2 and O3 values, and around 400 sensors, which should be updated hourly

Is a part of issue #542

### CODE REVIEW

File: adapters/southafrica.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  } catch (e) {
+    // Return undefined to be caught elsewhere
+    return undefined;
+  }
+  /**
+   * Convertor for pollutant, to make it look like the standard for the program, returns null if it is not an acceptable pollutant
+   * @param {String} pollutant that is going to be converted to the right format
+   * @returns {String} if pollutant is acceptable, null if not
+   */
+  var getPollutant = function (pollutant) {
```
[the remark this answers is not in the document]
Comment (magsyg) [reply adapters/southafrica.js:59]: The source has a lot of different parameters such as NOX, Amb Temp, WS, rain and such, and the getPollutant method should help with filtering out those values we do not want in the database, should I still replace it?

[the remark this answers is not in the document]
Comment (sruti) [reply adapters/southafrica.js:60]: Ah for that you can use [removeUnwantedParameters](https://github.com/openaq/openaq-fetch/blob/develop/lib/utils.js#L99)

File: adapters/southafrica.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      attribution: [{name: 'South African Air Quality Information System', url: 'http://saaqis.environment.gov.za'}],
+      averagingPeriod: {unit: 'hours', value: 1}
+    };
+    _.forOwn(s.monitors, function (v, key) {
+      if (v.value !== null) {
+        var m = _.clone(base);
+        m.parameter = v.Pollutantname;
+        m.value = Number(v.value);
+        m.unit = v.unit;
+        m.date = parseDate(v.DateVal);
```
Comment (sruti) [inline adapters/southafrica.js:82]: Looks like sometimes `dateVal` comes back `null`. For example: 
```
{"IP":null,"display":true,"STA_O3Factor":1.0,"STA_O3BFactor":0.0,"serialCode":304,"name":"Driftsands","location":"Driftsands Nature Reserve, Delft, Cape Town","city":"City of Cape Town","StGroup":null,"StGroupName":"Western Cape ","height":"37","latitude":"-33.991842","longitude":"18.653067","IndexColor":null,"IndexIcon":null,"IndexName":null,"environment":"N/A","GIS_Monitors":null,"GIS_channel":null,"GIS_Units":null,"GIS_LastRecivedValues":null,"timebase":60,"owner":"Western Cape Province","IndexValue":null,"Active":1,"UseMapView":1,"FileName":null,"ImageDescription":null,"ImageStatus":null,"NonContinuous":false,"AQSCODE":null,"STA_ImageData":null,"STA_Notes":"N/A","DateVal":null,"IndexValueDateTime":null,"STA_Target":"Urban","CBO_ItemIndex":2,"TargetId":18,"asix":null,"dateDisplay":"05/04/2021 10:00","monitors":[{"channel":2,"type":46,"value":"12.1","Lastvalue":"12.1","state":1,"name":"NO2","unit":"ppb","stationSerialCode":304,"stationName":"Driftsands","owner":"Western Cape Province","Color":null,"IndexIcon":null,"indexVal":0,"TimeBase":0,"StationTimeBase":60,"DateVal":null,"NumericFormat":5.3,"Pollutantname":"NO2","Description":"","asix":null,"isPolPos":false,"PollutantTimeBase":0,"PollutantID":1,"PctValid":75,"Active":1,"UseMapView":1,"isIndex":true,"indexName":null,"StationTag":"","ShortName":"Driftsands","County":"Western Cape"
```
On the website, it just reports it as `NO_REPORT_DATA` for the whole station so I would filter out any measurements that have null Date values

**Review by sruti — CHANGES_REQUESTED** [review]
So close! A few minor things and it will be good to go.

File: adapters/southafrica.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+   * @return {object} An object containing both UTC and local times
+   */
+  var parseDate = function (m) {
+    var date = moment.tz(m, 'YYYY-MM-DDHH:mm', 'Africa/Johannesburg');
+    return {utc: date.toDate(), local: date.format()};
+  };
+  var measurements = [];
+  _.forEach(data, function (s) {
+    const base = {
+      location: s.location,
```
[the remark this answers is not in the document]
Comment (sruti) [reply adapters/southafrica.js]: I actually meant for it to be `location: s.name` for all. I'll fix it

File: adapters/southafrica.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      city: s.city,
+      coordinates: {
+        latitude: Number(s.latitude),
+        longitude: Number(s.longitude)
+      },
+      attribution: [{name: 'South African Air Quality Information System', url: 'http://saaqis.environment.gov.za'}],
+      averagingPeriod: {unit: 'hours', value: 1}
+    };
+    _.forOwn(s.monitors, function (v, key) {
+      if (v.value !== null) {
```
[the remark this answers is not in the document]
Comment (sruti) [reply adapters/southafrica.js]: I might not have been clear in my explanation but wanted it to be ` if (v.value !== null && v.value !== '') {`

**Pull request merged**